src/exo/LoopIR_scheduling.py

```diff
@@ -3227,9 +3227,12 @@ def __init__(self, proc):
         super().__init__(proc)
 
         # might need to update IR with predicate changes
-        if new_preds := self.map_exprs(self.ir.preds):
-            # TODO KQ: is this line covered? do we not need to forward here?
-            self.ir = self.ir.update(preds=new_preds)
+        new_preds = self.map_exprs(self.ir.preds)
+        if new_preds:
+            self.ir, fwd = (
+                ic.Cursor.create(self.ir)._child_block("preds")._replace(new_preds)
```

Comment:
@SamirDroubi Do you often have cursors to predicates? If so, then they wouldn't forward under this implementation. If not, then maybe this is OK.

Reply:
No, I don't have cursors to predicates ever. The main issue was that other cursors weren't forwarding because of this.